src/core/CMakeLists.txt

```diff
@@ -142,7 +142,36 @@ set(EspressoCore_SRC
     virtual_sites/VirtualSitesRelative.cpp
 )
 
-add_library(EspressoCore SHARED ${EspressoCore_SRC})
+if(CUDA)
+    set(EspressoCuda_SRC
+        cuda_common_cuda.cu
+        cuda_init_cuda.cu
+        EspressoSystemInterface_cuda.cu
+        actor/DipolarBarnesHut_cuda.cu
+        actor/DipolarDirectSum_cuda.cu
+        actor/HarmonicOrientationWell_cuda.cu
+        actor/HarmonicWell_cuda.cu
+        actor/Mmm1dgpuForce_cuda.cu
+        electrostatics_magnetostatics/p3m_gpu_cuda.cu
+        electrostatics_magnetostatics/p3m_gpu_error_cuda.cu
+        grid_based_algorithms/electrokinetics_cuda.cu
+        grid_based_algorithms/lbgpu_cuda.cu
+        grid_based_algorithms/fd-electrostatics_cuda.cu
+        virtual_sites/lb_inertialess_tracers_cuda.cu
+        grid_based_algorithms/electrokinetics.cpp
+        grid_based_algorithms/electrokinetics_pdb_parse.cpp
+        grid_based_algorithms/lbgpu.cpp
+    )
+
+  set(CUDA_LINK_LIBRARIES_KEYWORD PUBLIC)
+  cuda_add_library(EspressoCore SHARED ${EspressoCore_SRC} ${EspressoCuda_SRC})
```

Comment:
Not sure that's a good idea. If we ever want to get Espresso into the Ubuntu repo, it should only depend on CUDA at runtime, meaning that any libcuda-dependent libraries should be conditionally loaded at runtime.

Reply:
The files in `EspressoCore_SRC` and `EspressoCuda_SRC` are mutually depended, this is the only way to write this down in CMake without lying to the build system. If this really becomes a problem it has to be solved by actually disentangling the code.